docs/basics/api-object.md

```diff
@@ -10,3 +10,51 @@ It is possible to customize the prefix by providing a string value to the
 `--class-prefix` flag, or by passing `--no-class-prefix`. (Note: if no prefix is
 used for k8s resources, this may lead to conflicts - see
 https://github.com/cdk8s-team/cdk8s/issues/140).
+
+### Resource Names
+
+Every resource has a name that is autogenerated when the cdk8s app is synthesized.
+The name is the passed in string you assign to an object, prefixed by the name of the
+chart it is contained in, and suffixed by the hash of the construct path.
+
+For example, the following code:
+
+```ts
+import { App, Chart } from 'cdk8s';
+import * as kplus from 'cdk8s-plus-25';
+
+const app = new App();
+const chart = new MyChart(app, 'my-chart');
+
+new kplus.Deployment(chart, 'deployment', {
+  containers: [{ image: 'nginx' }]
+});
+
+app.synth();
+```
+
+Generates a `Deployment` with the name: `my-chart-deployment-c8c354dd`.
+
+In Kubernetes, only one resource of a given kind can have a given name at a time.
+Hashes are used to ensure that all resources have unique identifiers. However, they are
+not required and it may sometimes be undesirable to include the hash suffixes.
+
+To remove hashes from resource names, you can change the above code by setting the
+`ChartProps.disableResourceNameHashes` value to `true` like so:
+
+```ts
+import { App, Chart } from 'cdk8s';
+import * as kplus from 'cdk8s-plus-25';
+
+const app = new App();
+const chart = new MyChart(app, 'my-chart', {
+    disableResourceNameHashes: true,
+});
+
+new kplus.Deployment(chart, 'deployment', {
+  containers: [{ image: 'nginx' }]
+});
+
+app.synth();
+```
+Which would similarly generate a `Deployment` now with the name: `my-chart-deployment`.
```

Comment:
```suggestion
Which would similarly generate a `Deployment` with the name: `my-chart-deployment`. Note that disabling hashes introduces the possibility of name clashes. For example, the following code:

```ts
// think of this as a custom construct
const c = new Construct(this, 'Construct');

new kplus.Deployment(c, 'Deployment', {
  containers: [{ image: 'image' }]
});

new kplus.Deployment(this, 'Construct-Deployment', {
  containers: [{ image: 'image' }]
})
```

Will produce two deployments with the same name: `chart-construct-deployment`. This happens because cdk8s uses a hyphen (`-`) as the delimiter between elements in the path, so if the id of your construct also contains a hyphen (`-`), a collision can occur.

In addition, you can explicitly set a name for every resource by using the `metadata.name` property:

```ts
new kplus.Deployment(c, 'Deployment', {
  metadata: {
    name: 'my-deployment'
  },
  ...
});
```

```